fix(kmodal): dismiss icon logic [beta] (#738)
Kong/kongponents: docs/components/modal.md

Use this to hide the built-in cancel button (`false` by default).

### hideDismissIcon

> When using the `header-image` slot we display a dismiss 'X' button in the upper right corner of the dialog. Set this prop to `true` to hide the.

Reviewer: this section is not complete. 

When using the `header-image` slot we display a dismiss 'X' button in the upper right corner of the dialog. Set this prop to `true` to hide the button (`false` by default).

Reviewer 2: @kaiarrowood these will need to go to `main` as well 😬

> When using the `header-image` slot we display a dismiss 'X' button in the upper right corner of the dialog. Set this prop to `true` to hide the.

Author: ```suggestion
When using the `header-image` slot we display a dismiss 'X' button in the upper right corner of the dialog. Set this prop to `true` to hide the button (`false` by default).
```

> When using the `header-image` slot we display a dismiss 'X' button in the upper right corner of the dialog. Boolean for hiding this button (`false` by default).

Author: ```suggestion
When using the `header-image` slot we display a dismiss 'X' button in the upper right corner of the dialog. Set this prop to `true` to hide the.
```
